testing/e2e/process_test.go

```diff
@@ -176,21 +164,28 @@ type agentProcess struct {
 }
 
 type agentConfig struct {
-	APIPort           int
-	BackendURLs       []string
 	CustomAttributes  string
 	ID                string
 	Redact            []string
-	SocketPort        int
 	KeepaliveTimeout  int
 	KeepaliveInterval int
+	Organization      string
+	Environment       string
 }
 
 // newAgent abstracts the initialization of an agent process and returns a
 // ready-to-use agent or exit with a fatal error if an error occurred while
 // initializing it
 func newAgent(config agentConfig, sensuctl *sensuCtl, t *testing.T) (*agentProcess, func()) {
-	agent := &agentProcess{agentConfig: config}
+	agent := &agentProcess{agentConfig: config, backendURL: sensuctl.wsURL}
+	agent.APIPort = int(atomic.AddInt64(&agentPortCounter, 1))
+	agent.SocketPort = int(atomic.AddInt64(&agentPortCounter, 1))
```

Comment:
Ah, so this is how ports are going to be coordinated! More deterministic this way.